firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarAction.swift

```diff
@@ -10,22 +10,35 @@ class ToolbarAction: Action {
     let addressToolbarModel: AddressToolbarModel?
     let navigationToolbarModel: NavigationToolbarModel?
     let numberOfTabs: Int?
+    let isEnabled: Bool?
+    let navigateBack: Bool?
+    let navigateForward: Bool?
 
     init(addressToolbarModel: AddressToolbarModel? = nil,
          navigationToolbarModel: NavigationToolbarModel? = nil,
          numberOfTabs: Int? = nil,
+         isEnabled: Bool? = nil,
+         navigateBack: Bool? = nil,
+         navigateForward: Bool? = nil,
          windowUUID: WindowUUID,
          actionType: ActionType) {
         self.addressToolbarModel = addressToolbarModel
         self.navigationToolbarModel = navigationToolbarModel
         self.numberOfTabs = numberOfTabs
+        self.isEnabled = isEnabled
+        self.navigateBack = navigateBack
+        self.navigateForward = navigateForward
         super.init(windowUUID: windowUUID, actionType: actionType)
     }
 }
 
 enum ToolbarActionType: ActionType {
     case didLoadToolbars
     case numberOfTabsChanged
+    case backButtonStatus
+    case forwardButtonStatus
```

Comment:
I haven't reviewed the previous toolbar redux PRs + haven't been in those conversations, so may not have as much context. Curious to hear the decision behind making `numberOfTabsChanged`, `backButtonStatus`, `forwardButtonStatus` action types. These sound like state types. Maybe we can rename them better, also happy to get on a call to discuss live to better understand.

Reply:
Yes, I believe you're 100% correct.